core/sql/optimizer/BindRelExpr.cpp

```diff
@@ -717,7 +717,7 @@ static ItemExpr *intersectColumns(const RETDesc &leftTable,
     ItemExpr *leftExpr  = leftTable.getValueId(i).getItemExpr();
     ItemExpr *rightExpr = rightTable.getValueId(i).getItemExpr();
     BiRelat *compare = new (bindWA->wHeap())
-      BiRelat(ITM_EQUAL, leftExpr, rightExpr);
+      BiRelat(ITM_EQUAL, leftExpr, rightExpr, TRUE);
```

Comment:
I use CQD nested_joins 'off' 
and it generates a HASH JOIN plan, and the test still works for null value. I don't know why, the plan for HASH_JOIN has an option 'u', not sure what does that mean.